packages/ui/ToggleItem/ToggleItem.js

```diff
@@ -15,25 +15,30 @@ import kind from '@enact/core/kind';
 import EnactPropTypes from '@enact/core/internal/prop-types';
 import PropTypes from 'prop-types';
 import compose from 'ramda/src/compose';
-import React from 'react';
+import React, {Fragment} from 'react';
 
 import ComponentOverride from '../ComponentOverride';
+import Slottable from '../Slottable';
 import Toggleable from '../Toggleable';
 import Touchable from '../Touchable';
 
 import componentCss from './ToggleItem.module.less';
 
 // eslint-disable-next-line
-const iconCreator = (position) => ({disabled, icon, iconComponent, selected, iconPosition}) => {
+const iconCreator = (position) => ({beforeItem, disabled, icon, iconComponent, iconPosition, selected}) => {
 	if (iconPosition === position) {
 		return (
-			<ComponentOverride
-				component={iconComponent}
-				disabled={disabled}
-				selected={selected}
-			>
-				{icon}
-			</ComponentOverride>
+			<Fragment>
+				{iconPosition === 'before' && beforeItem ? beforeItem : null}
```

Comment:
I think this and the part below can be simplified to just the following:
```
{iconPosition === 'before' && beforeItem}
```
Was there a case you found that I didn't account for?